kaldilm/csrc/arpa_lm_compiler.cc

```diff
@@ -203,10 +224,27 @@ void ArpaLmCompilerImpl<HistKey>::ConsumeNGram(const NGram &ngram,
     return;
   }
 
-  StateId source = source_it->second;
+  StateId source = source_it->second.state;
   StateId dest;
   Symbol sym = ngram.words.back();
   float weight = -ngram.logprob;
+  float backoff = -ngram.backoff;
+
+  if (low_order == nullptr) {
```

Comment:
is it necessary that the lm to subtract is actually of lower order?  If it would work regardless, we might as well make it as general as possible and name it accordingly.

Reply:
... if it can be done elegantly, we can consider generaly linear combinations of logprobs, not just one minus the other.